pallets/messages/src/lib.rs

```diff
@@ -138,24 +146,23 @@ pub mod pallet {
 
 			// TODO: validate schema existence and validity from schema pallet
 
-			let mut block_messages = <BlockMessages<T>>::get();
+			let mut block_messages = <BlockMessages<T>>::get().into_inner();
```

Comment:
```







```
BlockMessage::try_mutate(|messages| -> DispatchResult {
   messages.try_push(message).map_err(|_| Error::<T>::TooManyMessagesInBlock)?
  Ok(())
})?
```
Also, why do we need to store the current size in each element? I don't think we are using that index?

Reply:
Size is for storing total order. I will modify to use `try_mutate`